ai-prompt-flow/editorextensions/ai-prompt-flow/src/leftpanel.ts

```diff
@@ -0,0 +1,425 @@
+                } else if (block instanceof TableBlockProxy) {
+                    this.setupTableBlock(block);
+                    if (message.shape === 'examples') {
+                        //Connect it to the nearest system prompt block
+                        const nearest = nearestSystemPromptBlock(block.getPage(), block.getBoundingBox());
+                        this.setupExampleBlock(block, nearest);
+                    } else if (message.shape === 'variables') {
+                        const nearest = nearestUserPromptBlock(block.getPage(), block.getBoundingBox());
+                        this.setupVariablesBlock(block, nearest);
+                    }
+                    this.setupTableBlockStyle(block);
+                }
+
+                block?.setShadow({angle: 270, blur: 16, color: '#00000033', distance: 3});
+
+                await this.sendMessage({
+                    'message': 'dragDone',
+                });
+
+                // Update the right panel after the automatic connections have been made, to ensure
+                // the right panel is up to date with the state of the canvas.
+                await this.rightPanel.sendSelectionToPanel();
+            }
+        } else if (message.message == 'pointermove') {
+            this.viewport.dragPointerMove(message.x + this.framePosition.x, message.y + this.framePosition.y);
+        } else if (message.message == 'pointerup') {
+            this.viewport.dragPointerUp(message.x + this.framePosition.x, message.y + this.framePosition.y);
+        } else if (message.message == 'cancelDrag') {
+            this.viewport.cancelDraggingNewBlock();
+        } else if (message.message == 'requestEndpoints') {
+            const endpoints = await this.endpointsProxy.refreshAndGetAllEndpoints();
+            await this.sendMessage({
+                'endpoints': endpoints,
+            });
+        } else if (message.message == 'updateEndpoints') {
+            await this.endpointsProxy.setEndpoints(message.endpoints);
+            // ensure model selection in the right panel is up to date
+            await this.rightPanel.sendSelectionToPanel();
+        } else if (message.message == 'checkCanEditPackageSettings') {
+            const canEditPackageSettings = await this.client.canEditPackageSettings();
+            await this.sendMessage({
+                'message': 'canEditPackageSettings',
+                'canEditPackageSettings': canEditPackageSettings,
+            });
+        } else if (message.message == 'feedback') {
+            const beaconEvent = {
+                'name': 'sentAiFeedback',
+                'aiInteractionType': 'used ai prompt flow',
+                'rating': message.rating >= 4 ? 'thumbs up' : 'thumbs down',
+                'comment': 'Score: ' + message.rating + '/5. Comment: ' + message.feedback,
+            };
+            this.client.sendCommand(CommandName.FireBeaconEvent, {'e': JSON.stringify(beaconEvent)});
```

Comment:
We should replace this with `console.log(...)` or similar. The `FireBeaconEvent` command is only going to work for first-party extensions.